jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerSessionContext.kt

```diff
@@ -162,135 +170,249 @@ data class CodeModernizerSessionContext(
-                "./mvnw"
+            LOG.info { "Executing $mvnw" }
+            val installOutput = runInstallCommand(mvnw)
+            if (installOutput.exitCode != 0) {
+                LOG.error { "$mvnw install output: $installOutput" }
+                val error = "Maven Install: The exitCode should be 0 while it was ${installOutput.exitCode}"
+                CodetransformTelemetry.mvnBuildFailed(
+                    codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+                    codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
+                    reason = error
+                )
             }
-            val output = runCommand(mvnw)
-            if (output.exitCode != 0) {
-                LOG.error { "mvnw command output:\n$output" }
-                val error = "The exitCode should be 0 while it was ${output.exitCode}"
+            // TODO: currently running copy dependencies even if install failed.
+            //  This Should be updated to fast fail the transform if install failed.
+            val copyOutput = runCopyCommand(mvnw)
+            if (copyOutput.exitCode != 0) {
+                LOG.error { "$mvnw command output: $copyOutput" }
+                val error = "Maven Copy: The exitCode should be 0 while it was ${copyOutput.exitCode}"
                 CodetransformTelemetry.mvnBuildFailed(
                     codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                     codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
                     reason = error
                 )
-                return null
             } else {
-                LOG.warn { "mvnw executed successfully" }
+                LOG.info { "$mvnw executed successfully" }
                 shouldTryMvnCommand = false
             }
         } catch (e: ProcessNotCreatedException) {
-            val error = "./mvnw failed to execute as its likely not a unix machine"
+            val error = "$mvnw failed to execute as its likely not a unix machine"
+            buildlogBuilder.appendLine("$mvnw failed to execute: $e")
             CodetransformTelemetry.mvnBuildFailed(
                 codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                 codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
                 reason = error
             )
-            LOG.warn { error }
+            LOG.warn(e) { error }
         } catch (e: Exception) {
+            val error = "Unexpected exception when running $mvnw"
+            buildlogBuilder.appendLine("$mvnw failed to execute: $e")
             CodetransformTelemetry.mvnBuildFailed(
                 codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
                 codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.Mvnw,
-                reason = e.message
+                reason = error
```

Comment:
Does VSCode have the same error message?

Reply:
VS Code looks more like `MavenCopyDependenciesExecutionError`

Reply:
Possibly revisit how VSCode does and decide if being different is ok.

Reply:
Keeping the error message unchanged for now since the catch block is for all generic exception, which can be thrown during the execution of either install or copy.